lib/game.rb

```diff
@@ -0,0 +1,122 @@
+require "pry"
+class Game
+  attr_accessor :player_1, :player_2, :board
+
+WIN_COMBINATIONS = [
+  [0, 1, 2],
+  [3, 4, 5],
+  [6, 7, 8],
+  [0, 3, 6],
+  [1, 4, 7],
+  [2, 5, 8],
+  [0, 4, 8],
+  [6, 4, 2]
+]
+
+  def initialize(player_1=Player::Human.new("X"), player_2=Player::Human.new("O"), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+def turn
+  puts "Please enter 1-9:"
+  move = self.current_player.move(@board)
+  if @board.valid_move?(move)
+    @board.update(move, self.current_player)
+    @board.display
+  else
+    puts "That was an invalid move!"
+    turn
+  end
+end
+
+  def play
+    until over?
+      turn
+      won?
+    end
+
+    if won?
+      puts "Congratulations #{winner}!"
+    elsif draw?
+      puts "Cats Game!"
+    end
+  end
+
+def won?
+    WIN_COMBINATIONS.any? do |arr|
+      win_index_1 = arr[0]
+      win_index_2 = arr[1]
+      win_index_3 = arr[2]
+
+      position_1 = @board.cells[win_index_1]
+      position_2 = @board.cells[win_index_2]
+      position_3 = @board.cells[win_index_3]
+
+      if position_1 == "X" && position_2 == "X" && position_3 == "X" || 
+         position_1 == "O" && position_2 == "O" && position_3 == "O"
+         return arr
+      else
+        false
+      end
+    end
+end
+
+def draw?
+  if !(won?) && @board.full?
+    true
+  elsif !won? && !@board.full?
+    false
+  elsif won?
+    false
+  end
+end
+
+def over?
+  (won? || draw? || @board.full?) ? true : false
+end
+
+def winner
+  if !won?
+    nil
+  elsif won?
+    win = won?
+    char = @board.cells[win[0]]
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
elsif won = won?
  char = @board.cells[won[0]]
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.